src/zh/onemanhua/src/eu/kanade/tachiyomi/extension/zh/onemanhua/Onemanhua.kt

```diff
@@ -124,9 +124,12 @@ class Onemanhua : ConfigurableSource, ParsedHttpSource() {
     // Filter
     private class StatusFilter : Filter.TriState("已完结")
     private class SortFilter : Filter.Select<String>("排序", arrayOf("更新日", "收录日", "日点击", "月点击"), 2)
-
+    private class CategoryFilter : Filter.Select<String>("类型", arrayOf("全部", "热血", "玄幻", "恋爱", "冒险", "古风", "都市", "穿越", "奇幻", "其他", "少男", "搞笑", "战斗", "冒险热血", "重生", "爆笑", "逆袭", "后宫", "少年", "少女", "熱血", "系统", "动作", "校园", "冒險", "修真", "修仙", "剧情", "霸总", "大女主", "生活"), 0)
+    private class CharFilter : Filter.Select<String>("字母", arrayOf("全部", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"), 0)
```

Comment:
You can create a new `Filter.Select<String>` subclass to bind the display name to their respective category IDs:

```kotlin
open class UriPartFilter(displayName: String, private val vals: Array<Pair<String, String>>) :
    Filter.Select<String>(displayName, vals.map { it.first }.toTypedArray()) {
    fun toUriPart() = vals[state].second
}

private class CategoryFilter : UriPartFilter(
    "类型",
    arrayOf(
        "全部" to "",
        "热血" to "10023",
         // so on...
    )
)
```

And then, inside `searchMangaRequest`:
```kotlin
when (filter) {
   is CategoryFilter -> if (filter.state != 0) {
       url.addQueryParameter("mainCategoryId", filter.toUriPart())
   }
   // so on...
}
```